### PR TITLE
[16.0][IMP] product_supplierinfo_for_customer_sale: Add compatibility with Form() to change line name properly

### PR DESCRIPTION
**Pull request by victoralmau**
FWP from 15.0: https://github.com/OCA/sale-workflow/pull/3134 + https://github.com/OCA/sale-workflow/pull/3158

Add compatibility with Form() to change line name properly

@Tecnativa TT49073

### CODE REVIEW

File: product_supplierinfo_for_customer_sale/models/sale_order_line.py [hunk 1]
```diff
@@ -25,38 +27,30 @@ def _compute_product_customer_code(self):
                 code = ""
             line.product_customer_code = code
 
+    def _compute_name(self):
+        """Sobrescribimos el compute para re-definir el name."""
```
Comment (pedrobaeza) [inline product_supplierinfo_for_customer_sale/models/sale_order_line.py]: This comment should be in English and be more descriptive. What you are stating here is the what, which is clear - a method call `_compute_name` does that, to compute the name -, not the why.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Please put 2 commits, one with the changes as they were in v1, and another with the v16 adaptations.

**Comment by pedrobaeza** [comment]
The thumb rule for squashing in the same commit or not is if the changes needed in the new version are relevant enough to be kept separated. In this case, that the methods have totally changed in this new version, they are.

File: product_supplierinfo_for_customer_sale/models/sale_order_line.py (hunk 1 above)
Comment (victoralmau) [reply product_supplierinfo_for_customer_sale/models/sale_order_line.py]: Sorry, the comment has been changed.

**Comment by victoralmau** [comment]
Changes done (v15 commits + v16 commit)

**Review by pedrobaeza — APPROVED** [review]
Merging as done in previous version:

/ocabot merge patch

**Pull request merged**